cli/src/main/java/com/devonfw/tools/ide/tool/graalvm/GraalVm.java

```diff
@@ -0,0 +1,65 @@
+package com.devonfw.tools.ide.tool.graalvm;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Set;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.environment.EnvironmentVariables;
+import com.devonfw.tools.ide.environment.EnvironmentVariablesType;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+/**
+ * {@link LocalToolCommandlet} for <a href="https://www.graalvm.org/">GraalVM</a>, an advanced JDK with ahead-of-time
+ * Native Image compilation.
+ */
+public class GraalVm extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public GraalVm(IdeContext context) {
+
+    super(context, "graalvm", Set.of(Tag.JAVA));
+  }
+
+  @Override
+  public Path getToolPath() {
+
+    return this.context.getSoftwarePath().resolve("extra").resolve(getName());
+  }
+
+  @Override
+  public VersionIdentifier getInstalledVersion() {
+
+    return super.getInstalledVersion(getToolPath());
+  }
+
+  @Override
+  public void run() {
+
+    Path toolPath = getToolPath();
+    try {
+      if (!Files.exists(toolPath)) {
+        Files.createDirectories(toolPath);
+      }

```

Comment:
```suggestion
    try {
      if (!Files.exists(toolPath)) {
        Files.createDirectories(toolPath);
        super.install(true);
      }
```

Reply:
Probably you could nest the installation here, so it is triggered just if the tool is not installed I suppose... otherwise the installation process gets started every time the commandlet is run 😺

Reply:
I think it has to be outside of this if-condition. Otherwise it's not possible to install another update/version after the first installation.